File: src/tests/test_parser/TransitiveAndFiltering.py
from JavaParserVisitor import ClassInfo, MethodInfo

def expectedAnswer():
    ans = dict()
    transitiveAndFiltering = ClassInfo("TransitiveAndFiltering")
    transitiveAndFiltering.possibleFather = "Father"
    for letter in ["x", "y", "z"]:
        transitiveAndFiltering.members.add(letter)

    transitiveAndFiltering.methods["sum"] = MethodInfo("sum")
    transitiveAndFiltering.methods["sum"].memberReferences.add("x")

    transitiveAndFiltering.methods["invoker"] = MethodInfo("invoker")
    transitiveAndFiltering.methods["invoker"].memberReferences.add("x")
    transitiveAndFiltering.methods["invoker"].methodReferences.add("sum")

    transitiveAndFiltering.methods["rec"] = MethodInfo("rec")
    transitiveAndFiltering.methods["rec"].memberReferences.add("x")
    transitiveAndFiltering.methods["rec"].methodReferences.add("rec")


    ans["TransitiveAndFiltering"] = transitiveAndFiltering

    father = ClassInfo("Father")
    father.members.add("a")
    father.methods["foo"] = MethodInfo("foo")

    ans["Father"] = father

    return ans

File: src/tests/test_parser/TransitiveAndFiltering.java
public class Father {
  int a;
  public void foo() {}
}

public class TransitiveAndFiltering extends Father {
  public int sum() {
    return x + a;
  }
  public void invoker() {
    sum();
    foo();
    someRandomThing.invoker();
  }
  public void rec() {
    this.rec();
    x++;
    // y++;
  }

  int x;
  int y;
  int z;
}